client/components/data/query-email-forwards/index.jsx

```diff
@@ -0,0 +1,55 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import isRequestingEmailForwards from 'state/selectors/is-requesting-email-forwards';
+import { getEmailForwards } from 'state/email-forwarding/actions';
+
+class QueryEmailForwards extends PureComponent {
+	static propTypes = {
+		domainName: PropTypes.string.isRequired,
+		getEmailForwards: PropTypes.func.isRequired,
+		requestingEmailForwards: PropTypes.bool.isRequired,
+	};
+
+	componentDidMount() {
+		this.request();
+	}
+
+	request() {
+		if ( ! this.props.requestingEmailForwards ) {
+			this.props.getEmailForwards( this.props.domainName );
+		}
+	}
+
+	render() {
+		return null;
+	}
+}
+
+export default connect(
+	( state, ownProps ) => {
```

Comment:
this could be simplified to an object-returning arrow that uses ownProps directly:
```
=> ( { requestingEmailForwards: isRequestingEmailForwards( state, ownProps.domainName ) } )
```